### PR TITLE
Represent package+module according to Kythe schema (fixes #8).

### LINKED ISSUE #8: Export Haskell packages and modules
Take (Cabal package) + (Haskell module) = Kythe package
Emit to index:

- defines/binding from module name to package node
- childof from top-level decls to module node

Note: import refs to the module are in separate issue, see below.

### PR DESCRIPTION
**Pull request by robinp**
See https://kythe.io/docs/schema/#package and the added verifier test for the emitted entities.

### CODE REVIEW

**Comment by robinp** [comment]
+@ivan444 to review.

File: kythe-verification/README.md [hunk 1]
```diff
@@ -0,0 +1,9 @@
+# Kythe verification
+
+The Kythe verifier parses expectation-annotated source code, and correlates it
+with the Kythe entries generated from that source.
+See [Kythe verifier docs](http://kythe.io/docs/kythe-verifier.html) and
+[testing a new indexer](http://kythe.io/docs/schema/writing-an-indexer.html#_testing).
+
+Aim to keep the verifier sources small, as the unification gets slow pretty fast.
+Verify only the aspects you care about.
```
Comment (ivan444) [inline kythe-verification/README.md]: Maybe also include the instructions how to run the verifier? (e.g., it's run by test.sh, you must run stack install, you must be in the verifier dir, the kythe location is controlled by VERIFIER env var, etc.).

File: haskell-indexer-translate/src/Language/Haskell/Indexer/Translate.hs [hunk 2]
```diff
@@ -30,6 +30,7 @@ module Language.Haskell.Indexer.Translate
     --
     , XRef(..)
     , AnalysedFile(..)
+    , ModuleTick(..)
```
Comment (ivan444) [inline haskell-indexer-translate/src/Language/Haskell/Indexer/Translate.hs]: Is Module a reserved name? The rest of entities don't have "Tick" suffix. Maybe remove it for consistency?

File: haskell-indexer-backend-ghc/src/Language/Haskell/Indexer/Backend/Ghc.hs [hunk 3]
```diff
@@ -166,6 +170,12 @@ modifyDecl declMods decl =
     applyMod (MethodForInstance i) =
         withExtra (\e -> e {methodForInstance = Just $! i}) decl
 
+-- | Bundles up the module name with its source span.
+mkModuleTick :: (Given ExtractCtx) => ParsedSource -> PkgModule -> ModuleTick
+mkModuleTick lhsm pm = ModuleTick pm span
+  where
+    span = (fmap getLoc . hsmodName . unLoc $ lhsm) >>= srcSpanToSpan
```
Comment (ivan444) [inline haskell-indexer-backend-ghc/src/Language/Haskell/Indexer/Backend/Ghc.hs]: Maybe rename "span" as it's shadowing Prelude function?

File: kythe-verification/README.md (hunk 1 above)
Comment (robinp) [inline kythe-verification/README.md]: Done.

File: haskell-indexer-translate/src/Language/Haskell/Indexer/Translate.hs (hunk 2 above)
Comment (robinp) [inline haskell-indexer-translate/src/Language/Haskell/Indexer/Translate.hs]: It's analogous to `Tick` which is a unique name for something. I agree it's not best. `Module` happens in GHC, so I would avoid the confusion if possible.

File: haskell-indexer-backend-ghc/src/Language/Haskell/Indexer/Backend/Ghc.hs (hunk 3 above)
Comment (robinp) [inline haskell-indexer-backend-ghc/src/Language/Haskell/Indexer/Backend/Ghc.hs]: Done.

**Comment by robinp** [comment]
PTAL.

**Pull request merged**